src/aap_eda/services/project/scm.py

```diff
@@ -322,3 +327,32 @@ def __call__(
                     raise ScmAuthenticationError(err_msg)
                 raise ScmError(f"{self.ERROR_PREFIX} {err_msg}")
             raise ScmError(f"{self.ERROR_PREFIX} {outputs.getvalue().strip()}")
+
+
+URL_REGEX = "^((git|ssh|http(s)?)|git\\+ssh|(git@[\\w\\.-]+))(:(//)?)([\\w\\.@\\:/\\-~]+)(\\.git)?(/)?$"  # noqa: E501
```

Comment:
@bzwei should this be prefixed with r"...."

Reply:
It is always better to compile the expression for performance reasons. 
Docs: https://docs.python.org/3/library/re.html#re.compile 
Example: https://github.com/ansible/eda-server/blob/e6e49f146ace9fdc9d220de041017736613dd720/src/aap_eda/core/utils/rulebook_process_logs.py#L21